omnigibson/maps/traversable_map.py

```diff
@@ -226,30 +175,36 @@ def get_shortest_path(self, floor, source_world, target_world, entire_path=False
             source_world (2-array): (x,y) 2D source location in world reference frame (metric)
             target_world (2-array): (x,y) 2D target location in world reference frame (metric)
             entire_path (bool): whether to return the entire path
+            robot (None or BaseRobot): if given, erode the traversability map to account for the robot's size
 
         Returns:
             2-tuple:
                 - (N, 2) array: array of path waypoints, where N is the number of generated waypoints
                 - float: geodesic distance of the path
         """
-        assert self.build_graph, "cannot get shortest path without building the graph"
         source_map = tuple(self.world_to_map(source_world))
         target_map = tuple(self.world_to_map(target_world))
 
-        g = self.floor_graph[floor]
-
-        if not g.has_node(target_map):
-            nodes = np.array(g.nodes)
-            closest_node = tuple(nodes[np.argmin(np.linalg.norm(nodes - target_map, axis=1))])
-            g.add_edge(closest_node, target_map, weight=T.l2_distance(closest_node, target_map))
+        # create a deep copy so that we don't erode the original map
+        trav_map = copy.deepcopy(self.floor_map[floor])
 
-        if not g.has_node(source_map):
-            nodes = np.array(g.nodes)
-            closest_node = tuple(nodes[np.argmin(np.linalg.norm(nodes - source_map, axis=1))])
-            g.add_edge(closest_node, source_map, weight=T.l2_distance(closest_node, source_map))
-
-        path_map = np.array(nx.astar_path(g, source_map, target_map, heuristic=T.l2_distance))
+        # erode the traversability map to account for the robot's size
+        if robot:
+            if hasattr(robot, 'tucked_aabb_extent'):
+                robot_chassis_extent = robot.tucked_aabb_extent[:2]
+            else:
+                robot_chassis_extent = robot.aabb_extent[:2]
+            robot_radius = np.linalg.norm(robot_chassis_extent) / 2.0
+            robot_radius_pixel = int(np.ceil(robot_radius / self.map_resolution))
+            trav_map = cv2.erode(trav_map, np.ones((robot_radius_pixel, robot_radius_pixel)))
+        else:
+            # convert default_erosion_radius from meter to pixel
+            erosion_radius_pixel = int(np.ceil(self.default_erosion_radius / self.map_resolution))
+            trav_map = cv2.erode(trav_map, np.ones((erosion_radius_pixel, erosion_radius_pixel)))
 
+        path_map = astar(trav_map, source_map, target_map)
+        if path_map is None:
+            raise ValueError("No path found")
```

Comment:
Perhaps a bit more descriptive error message -- e.g.: contextualizing that it was trying to find a traversable path, etc.